docs/core/extensions/httpclient-factory-troubleshooting.md

```diff
@@ -0,0 +1,184 @@
+
+For more information, see the [Message Handler Scopes in IHttpClientFactory](httpclient-factory.md#message-handler-scopes-in-ihttpclientfactory) section in the `IHttpClientFactory` guidelines.
+
+## `HttpClient` doesn't respect DNS changes
+
+Even if `IHttpClientFactory` is used, it is still possible to hit the stale DNS problem. This can usually happen if an `HttpClient` instance gets captured in a `Singleton` service, or, in general, stored somewhere for the time longer than the specified `HandlerLifetime`. `HttpClient` will also get captured if the respective _typed client_ gets captured by a singleton.
+
+❌ DO NOT cache `HttpClient` instances created by `IHttpClientFactory` for prolonged periods of time.
+
+❌ DO NOT inject _typed client_ instances into `Singleton` services.
+
+✔️ CONSIDER requesting a client from `IHttpClientFactory` in a timely manner or each time you need one. Factory-created clients are are safe to dispose.
+
+`HttpClient` instances created by `IHttpClientFactory` are intended to be **short-lived**.
+
+- Recycling and recreating `HttpMessageHandler`'s when their lifetime expires is essential for `IHttpClientFactory` to ensure the handlers react to DNS changes. `HttpClient` is tied to a specific handler instance upon its creation, so new `HttpClient` instances should be requested in a timely manner to ensure the client will get the updated handler.
+
+- Disposing of such `HttpClient` instances **created by the factory** will not lead to socket exhaustion, as its disposal **will not** trigger disposal of the `HttpMessageHandler`. `IHttpClientFactory` tracks and disposes of resources used to create `HttpClient` instances, specifically the `HttpMessageHandler` instances, as soon their lifetime expires and there's no `HttpClient` using them anymore.
+
+_Typed clients_ are intended to be **short-lived** as well, as an `HttpClient` instance is injected into the constructor, so it will share the _typed client_ lifetime.
+
+For more information, see the [`HttpClient` lifetime management](httpclient-factory.md#httpclient-lifetime-management) and [Avoid _typed clients_ in singleton services](httpclient-factory.md#avoid-typed-clients-in-singleton-services) sections in the `IHttpClientFactory` guidelines.
+
+## `HttpClient` uses too many sockets
+
+Even if `IHttpClientFactory` is used, it is still possible to hit the socket exhaustion issue with a specific usage scenario. By default, `HttpClient` doesn't limit the amount of concurrent requests. If a large amount of HTTP/1.1 requests are started concurrently at the same time, each of them will end up triggering a new HTTP connection attempt, because there is no free connection in the pool and no limit is set.
```

Comment:
```suggestion
Even if `IHttpClientFactory` is used, it is still possible to hit the socket exhaustion issue with a specific usage scenario. By default, `HttpClient` doesn't limit the number of concurrent requests. If a large number of HTTP/1.1 requests are started concurrently at the same time, each of them will end up triggering a new HTTP connection attempt, because there is no free connection in the pool and no limit is set.
```